ttnn/cpp/ttnn/tensor/layout/tensor_layout.cpp

```diff
@@ -122,29 +151,82 @@ size_t TensorLayout::compute_page_size_bytes(const Size& page_size) const {
     return page_config_.get_page_size_bytes(page_size, dtype_);
 }
 
+Size TensorLayout::compute_logical_shard_shape(const ttnn::SimpleShape& shape) const {
+    if (shard_shape_.has_value()) {
+        return shard_shape_.value();
+    }
+    const int rank = static_cast<int>(shape.rank());
+
+    size_t shard_width = 1;
+    size_t shard_height = 1;
+
+    // Iterate dims in reverse order
+    // Even tensor of rank 0 or 1
+    for (int i = -1; i >= -2; --i) {
```

Comment:
Should we continue to `-rank`? 
Otherwise the code is equivalent to:
```
size_t shard_width = shape[-1];
size_t shard_height = shape[-2]
```

Reply:
Yea, I can change it to make it clearer. The only thing this covers is rank = 0 or 1, but otherwise it's equivalent.